src/ServiceManagement/Profile/Commands.Profile/Account/AddAzureAccount.cs

```diff
@@ -90,7 +121,64 @@ public override void ExecuteCmdlet()
                 }
 
                 WriteObject(account.ToPSAzureAccount());
-            } 
+            }
+        }
+
+        private IAzureAccount GetAccountFromSPN(AzureAccount account, IAzureEnvironment environment, SecureString password)
+        {
+            IAzureAccount tenantAccount = new AzureAccount();
+            CopyAccount(account, tenantAccount);
+            WriteDebugWithTimestamp("AzureSession.Instance.AuthenticationFactory.Authenticate");
+            var tenantToken = AzureSession.Instance.AuthenticationFactory.Authenticate(
+                tenantAccount,
+                environment,
+                Tenant,
+                password,
+                ShowDialog.Never,
+                null);
+            if (string.Equals(tenantAccount.Id, account.Id, StringComparison.InvariantCultureIgnoreCase))
+            {
+                tenantAccount = account;
+            }
+            tenantAccount.SetOrAppendProperty(AzureAccount.Property.Tenants, new string[] { Tenant });
+            WriteDebugWithTimestamp("Create mangement client");
+            using (var managementClient = AzureSession.Instance.ClientFactory.CreateCustomClient<ManagementClient>(
+                            new TokenCloudCredentials(SubscriptionId, tenantToken.AccessToken),
+                            environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ServiceManagement)))
+            {
+                WriteDebugWithTimestamp("Get Subscription");
+                var subscription = managementClient.Subscriptions.Get();
+
+                AzureSubscription psSubscription = new AzureSubscription
+                {
+                    Id = subscription.SubscriptionID,
+                    Name = subscription.SubscriptionName,
+                };
+                psSubscription.SetEnvironment(environment.Name);
+                psSubscription.SetProperty(AzureSubscription.Property.Tenants,
+                    Tenant);
+                psSubscription.SetAccount(tenantAccount.Id);
+                tenantAccount.SetOrAppendProperty(AzureAccount.Property.Subscriptions,
+                    new string[] { psSubscription.Id });
+
+                WriteDebugWithTimestamp("AddOrSetAccount");
+                // Add account 
+                ProfileClient.AddOrSetAccount(tenantAccount);
+
+                WriteDebugWithTimestamp("AddOrSetSubscription");
+                ProfileClient.AddOrSetSubscription(psSubscription);
+
+                // Set subscription as Default
+                ProfileClient.SetSubscriptionAsDefault(psSubscription.Name, psSubscription.GetProperty(AzureSubscription.Property.Account));
```

Comment:
The function name is ```Get*```, but it's actually doing add/set/update operations, so consider having a more accurate function name.

Reply:
@huangpf  The cmdlet verb is actually "Add".  (Add-AzureAccount)